src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/IRequestBodyReader.cs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http
+{
+    public interface IRequestBodyReader
+    {
+        Task StartAsync(MessageBody messageBody, CancellationToken cancellationToken = default(CancellationToken));
```

Comment:
Might be better to make cancellationToken a required parameter. It will save us from accidentally not passing the token to methods as we work on this code.